projects/components/src/filtering/filter/filter.service.ts

```diff
@@ -0,0 +1,205 @@
+import { SplitFilter } from './parser/parsed-filter';
+import { AbstractFilterParser } from './parser/types/abstract-filter-parser';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FilterService {
+  private static readonly FILTER_QUERY_PARAM: string = 'filter';
+
+  public constructor(
+    private readonly navigationService: NavigationService,
+    private readonly filterBuilderLookupService: FilterBuilderLookupService,
+    private readonly filterParserLookupService: FilterParserLookupService
+  ) {}
+
+  /*
+   * URL Filtering
+   */
+
+  public getUrlFiltersChanges$(attributes: FilterAttribute[]): Observable<Filter[]> {
+    return this.navigationService.navigation$.pipe(map(() => this.getUrlFilters(attributes)));
+  }
+
+  public getUrlFilters(attributes: FilterAttribute[]): Filter[] {
+    return this.navigationService
+      .getAllValuesForQueryParameter(FilterService.FILTER_QUERY_PARAM)
+      .map(filterString => this.parseUrlFilterString(attributes, filterString))
+      .filter((filter): filter is Filter => filter !== undefined);
+  }
+
+  public setUrlFilters(filters: Filter[]): void {
+    this.navigationService.addQueryParametersToUrl({
+      [FilterService.FILTER_QUERY_PARAM]: filters.length === 0 ? undefined : filters.map(f => f.urlString)
+    });
+  }
+
+  public applyUrlFilter(attributes: FilterAttribute[], filter: Filter): void {
+    const filters = this.getUrlFilters(attributes);
+
+    const foundIndex = filters.findIndex(f => filter.field === f.field);
+
+    if (foundIndex !== -1) {
+      filters[foundIndex] = filter;
+    } else {
+      filters.push(filter);
+    }
+
+    this.setUrlFilters([...filters]);
+  }
+
+  public removeUrlFilter(attributes: FilterAttribute[], filter: Filter): void {
+    this.setUrlFilters([...this.getUrlFilters(attributes).filter(f => filter.field !== f.field)]);
+  }
+
+  private parseUrlFilterString(attributes: FilterAttribute[], filterString: string): Filter | undefined {
+    return attributes
+      .filter(attribute => this.filterBuilderLookupService.isBuildableType(attribute.type))
+      .flatMap(attribute => {
+        const filterBuilder = this.filterBuilderLookupService.lookup(attribute.type);
+        const supportedUrlOperators = filterBuilder.supportedOperators().map(toUrlFilterOperator);
+
+        const splitUrlFilter = AbstractFilterParser.splitFilterStringByOperator(
+          supportedUrlOperators,
+          filterString,
+          false
+        );
+
+        if (splitUrlFilter === undefined) {
+          return undefined;
+        }
+
+        const filterParser = this.filterParserLookupService.lookup(fromUrlFilterOperator(splitUrlFilter.operator));
+
+        const parsedFilter = filterParser.parseUrlFilterString(attribute, filterString);
+
+        if (parsedFilter === undefined) {
+          return undefined;
+        }
+
+        return splitUrlFilter.lhs === parsedFilter.field
```

Comment:
What's the scenario that could lead to this not being true?

Reply:
The only case right now is the new string map dot syntax. It requires further parsing.

That said, this method gets refactored in the following PR since I choose to keep _URL_ filters the original syntax and this check goes away. It's a handful of changes that work in concert, so not worth pulling them in here, but I can call it out for review in the next PR.